### PR TITLE
pipeline fixes for mass testing

### PR DESCRIPTION
**Pull request by drblallo**
introduces pipleine analyze all and fixes llvmcontainers merge back issues.

### CODE REVIEW

File: tools/pipeline/Main.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ static cl::list<string>
   InputPipeline("P", desc("<Pipeline>"), cat(PipelineCategory));
 
 static cl::list<string> Targets(Positional,
-                                OneOrMore,
+                                ZeroOrMore,
```
Comment (aleclearmind) [inline tools/pipeline/Main.cpp:40]: What happens if there's zero targets and no `--analyze-all`?

File: include/revng/Pipeline/LLVMContainer.h [hunk 2]
```diff
@@ -142,49 +143,79 @@ class LLVMContainerBase
     auto BeforeEnumeration = this->enumerate();
```
Comment (aleclearmind) [inline include/revng/Pipeline/LLVMContainer.h:143]: I'd put a `using namespace llvm`

File: include/revng/Pipeline/LLVMContainer.h [hunk 3]
```diff
@@ -142,49 +143,79 @@ class LLVMContainerBase
...
+    std::set<std::string> Internals;
+    std::set<std::string> Externals;
+
+    // All symbols privated and not must be transformed into weak symbols, so
+    // that when multiple with the same name exists, one is dropped.
+    for (auto &Global : ToMerge.getModule().global_objects()) {
+
+      if (Global.getLinkage() == llvm::GlobalValue::InternalLinkage) {
+        Internals.insert(Global.getName().str());
+        Global.setLinkage(llvm::GlobalValue::LinkOnceODRLinkage);
```
Comment (aleclearmind) [inline include/revng/Pipeline/LLVMContainer.h:162]: Please justify the choice of `LinkOnceODRLinkage`. I remember you had *reasons*.

File: include/revng/Pipeline/LLVMContainer.h [hunk 4]
```diff
@@ -142,49 +143,79 @@ class LLVMContainerBase
...
+        Global.setLinkage(llvm::GlobalValue::ExternalLinkage);
     }
 
-    revng_assert(llvm::verifyModule(*Composite, nullptr) == 0);
-    Module = std::move(Composite);
+    // We must ensure output is valid
+    revng_assert(llvm::verifyModule(*ToMerge.Module, nullptr) == 0);
+    Module = std::move(ToMerge.Module);
 
+    // checks that module merging commutes w.r.t. enumeration, as specified in
```
Comment (aleclearmind) [inline include/revng/Pipeline/LLVMContainer.h:217]: Mind the case

**Review by aleclearmind — CHANGES_REQUESTED** [review]

File: tools/pipeline/Main.cpp (hunk 1 above)
Comment (drblallo) [reply tools/pipeline/Main.cpp:40]: it terminates with great success

File: include/revng/Pipeline/LLVMContainer.h (hunk 2 above)
Comment (drblallo) [reply include/revng/Pipeline/LLVMContainer.h:143]: i don't think the use is so pervasive that we gain much from doing so

File: include/revng/Pipeline/LLVMContainer.h (hunk 3 above)
Comment (aleclearmind) [reply include/revng/Pipeline/LLVMContainer.h:149]: *cries in linker script*

**Comment by aleclearmind** [comment]
Merged in 1d337bd. Thanks!

**Pull request closed without merging**